dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobType.java

```diff
@@ -51,148 +62,126 @@
-  TRACKER_IMPORT_JOB(false),
-  TRACKER_IMPORT_NOTIFICATION_JOB(false),
-  TRACKER_IMPORT_RULE_ENGINE_JOB(false),
-  MATERIALIZED_SQL_VIEW_UPDATE(true, SchedulingType.CRON, SqlViewUpdateParameters.class, null),
-  LOCK_EXCEPTION_CLEANUP(true, SchedulingType.CRON, LockExceptionCleanupJobParameters.class, null),
-
-  // Internal jobs
-  LEADER_ELECTION(false),
-  LEADER_RENEWAL(false),
-  COMPLETE_DATA_SET_REGISTRATION_IMPORT(false),
-  DATAVALUE_IMPORT_INTERNAL(false),
-  METADATA_IMPORT(false),
-  DATAVALUE_IMPORT(false),
-  GEOJSON_IMPORT(false),
-  EVENT_IMPORT(false),
-  ENROLLMENT_IMPORT(false),
-  TEI_IMPORT(false),
-  DISABLE_INACTIVE_USERS(true, SchedulingType.CRON, DisableInactiveUsersJobParameters.class, null),
-  ACCOUNT_EXPIRY_ALERT(false),
-  SYSTEM_VERSION_UPDATE_CHECK(false),
-
-  // Testing purposes
-  TEST(true, SchedulingType.CRON, TestJobParameters.class, null),
-  MOCK(false, SchedulingType.CRON, MockJobParameters.class, null),
-
-  // Deprecated, present to satisfy code using the old enumeration
-  // TaskCategory
-  @Deprecated
-  GML_IMPORT(false),
-  @Deprecated
-  ANALYTICSTABLE_UPDATE(false),
-  @Deprecated
-  PROGRAM_DATA_SYNC(false);
-
-  private final boolean configurable;
-
-  private final SchedulingType schedulingType;
+  /*
+  User defined jobs
+   */
+  DATA_INTEGRITY(DataIntegrityJobParameters.class),
+  RESOURCE_TABLE(),
+  ANALYTICS_TABLE(AnalyticsJobParameters.class),
+  CONTINUOUS_ANALYTICS_TABLE(ContinuousAnalyticsJobParameters.class),
+  DATA_SYNC(DataSynchronizationJobParameters.class),
+  TRACKER_PROGRAMS_DATA_SYNC(TrackerProgramsDataSynchronizationJobParameters.class),
+  EVENT_PROGRAMS_DATA_SYNC(EventProgramsDataSynchronizationJobParameters.class),
+  META_DATA_SYNC(MetadataSyncJobParameters.class),
+  AGGREGATE_DATA_EXCHANGE(AggregateDataExchangeJobParameters.class),
+  SEND_SCHEDULED_MESSAGE(),
+  PROGRAM_NOTIFICATIONS(),
+  MONITORING(MonitoringJobParameters.class),
+  PUSH_ANALYSIS(PushAnalysisJobParameters.class),
+  TRACKER_SEARCH_OPTIMIZATION(TrackerTrigramIndexJobParameters.class),
+  PREDICTOR(PredictorJobParameters.class),
+  MATERIALIZED_SQL_VIEW_UPDATE(SqlViewUpdateParameters.class),
+  DISABLE_INACTIVE_USERS(DisableInactiveUsersJobParameters.class),
+  TEST(TestJobParameters.class),
+  LOCK_EXCEPTION_CLEANUP(
+      LockExceptionCleanupJobParameters.class,
+      daily2am("OQ9KeLgqy20", "Remove lock exceptions older than 6 months")),
+
+  /*
+  Programmatically used Jobs
+  */
+  MOCK(MockJobParameters.class),
+  SMS_SEND(SmsJobParameters.class),
+  TRACKER_IMPORT_JOB(),
+  TRACKER_IMPORT_NOTIFICATION_JOB(),
+  TRACKER_IMPORT_RULE_ENGINE_JOB(),
+  IMAGE_PROCESSING(),
+  COMPLETE_DATA_SET_REGISTRATION_IMPORT(),
+  DATAVALUE_IMPORT_INTERNAL(),
+  METADATA_IMPORT(),
+  DATAVALUE_IMPORT(ImportOptions.class),
+  GEOJSON_IMPORT(),
+  EVENT_IMPORT(),
+  ENROLLMENT_IMPORT(),
+  TEI_IMPORT(),
+  GML_IMPORT(),
+
+  /*
+  System Jobs
+  */
+  HEARTBEAT(every(20, "DHIS2rocks1", "Heartbeat")),
```

Comment:
providing hard-coded UIDs seems a bit strange here when we usually generate them everywhere else. Unless there's a specific reason for this?